ops/model.py

```diff
@@ -1400,6 +1414,27 @@ def get_pebble(self, socket_path: str) -> 'pebble.Client':
         """Create a pebble.Client instance from given socket path."""
         return pebble.Client(socket_path=socket_path)
 
+    def add_pebble_layer_file(self, layers_dir, layer_yaml, layer_name='layer'):
+        # Create the directory if it doesn't exist
+        os.makedirs(layers_dir, exist_ok=True)
+
+        # Fetch the sorted filenames of the current layers (e.g., "001-foo.yaml")
+        layers = sorted(n for n in os.listdir(layers_dir)
+                        if self.PEBBLE_LAYER_FILE_RE.match(n))
+
+        if layers:
+            # Find the last layer number
+            last = int(layers[-1].split('-', 1)[0])
+            assert 1 <= last <= 999, last
```

Comment:
assert in runtime code isn't great (IIRC it can get disabled sometimes, we should have a proper exception if we need one)
Also, this *specific* logic will overflow at 999, where it will read a last layer of 999, then add 1, and then write out a 1000 anyway.